### PR TITLE
etcd: Fix session renewal controllers

### PR DESCRIPTION
**Pull request by tgraf**

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

File: pkg/kvstore/etcd.go [hunk 1]
```diff
@@ -515,10 +531,12 @@ func (e *etcdClient) renewLockSession(ctx context.Context) error {
 	// routines can get a lease ID of an already expired lease.
 	e.Lock()
 
+	timeoutCtx, cancel := context.WithTimeout(e.client.Ctx(), statusCheckTimeout)
```
Comment (sayboras) [inline pkg/kvstore/etcd.go]: any reason why we can't use _ctx_ here (e.g. context.WithTimeout(ctx, statusCheckTimeout) ) ?

Comment (tgraf) [reply pkg/kvstore/etcd.go]: Good catch. Fixed.

**Review by qmonnet — APPROVED** [review]

File: pkg/kvstore/etcd.go [hunk 2]
```diff
@@ -658,32 +676,21 @@ func connectEtcdClient(ctx context.Context, config *client.Config, cfgPath strin
 
 	ec.controllers.UpdateController("kvstore-etcd-session-renew",
 		controller.ControllerParams{
+			// Stop controller function when etcd client is terminating
+			Context: ec.client.Ctx(),
```
Comment (joestringer) [inline pkg/kvstore/etcd.go:680]: :exploding_head:

**Review by joestringer — APPROVED** [review]

**Comment by tgraf** [comment]
test-me-please

**Comment by qmonnet** [comment]
Backport to v1.6 in #12748.

**Pull request merged**